StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -49,158 +51,161 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 
-                    goto default;
+                    haveLeadingSpace = !partOfUnaryExpression && !startOfIndexer && !consecutiveCast && !partOfInterpolation;
+                    haveTrailingSpace = false;
+                    break;
+
+                case SyntaxKind.ParameterList:
+                    var partOfLambdaExpression = token.Parent.Parent.IsKind(SyntaxKind.ParenthesizedLambdaExpression);
+                    haveLeadingSpace = partOfLambdaExpression;
+                    haveTrailingSpace = false;
+                    break;
 
                 default:
-                    if (precedingToken.Parent is BinaryExpressionSyntax
-                        || precedingToken.Parent is AssignmentExpressionSyntax
-                        || precedingToken.Parent is ConditionalExpressionSyntax
-                        || precedingToken.Parent is EqualsValueClauseSyntax)
-                    {
-                        allowLeadingNoSpace = false;
-                        allowLeadingSpace = true;
-                        break;
-                    }
-                    else
-                    {
-                        allowLeadingNoSpace = true;
-                        allowLeadingSpace = false;
-                        break;
-                    }
-                }
+                    haveLeadingSpace = false;
+                    haveTrailingSpace = false;
+                    break;
             }
 
-            followedBySpace = token.HasTrailingTrivia;
-            lastInLine = followedBySpace && token.TrailingTrivia.Any(SyntaxKind.EndOfLineTrivia);
-
-            if (reportPrecedingError && !firstInLine)
+            // Ignore spacing before if another opening parenthesis is before this.
+            // That way the first opening parenthesis will report any spacing errors.
+            if (!prevTokenIsOpenParen)
             {
-                if (!allowLeadingSpace && precededBySpace)
-                {
-                    // Opening parenthesis must{ not} be {preceded} by a space.
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, token.GetLocation(), " not", "preceded"));
-                }
-                else if (!allowLeadingNoSpace && !precededBySpace)
+                var hasLeadingComment = (leadingTriviaList.Count > 0) && leadingTriviaList.Last().IsKind(SyntaxKind.MultiLineCommentTrivia);
+                var hasLeadingSpace = (leadingTriviaList.Count > 0) && leadingTriviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+
+                if (!isFirstOnLine && !hasLeadingComment && (haveLeadingSpace ^ hasLeadingSpace))
                 {
-                    // Opening parenthesis must{} be {preceded} by a space.
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, token.GetLocation(), string.Empty, "preceded"));
+                    var properties = new Dictionary<string, string>
+                    {
+                        { "location", "preceding" },
```

Comment:
:bulb: Define `internal const` strings for these keys and values, and use them in the code fix. Also, consider using this:

```cs
var properties = ImmutableDictionary.Create<string, string>()
    .Add("location", "preceding")
    .Add("action", haveLeadingSpace ? "insert" : "remove");
```

Since `properties` will now be an `ImmutableDictionary<string, string>`, you can also remove the call to `ToImmutableDictionary()`.